napari/_qt/tests/test_qt_viewer.py

```diff
@@ -22,136 +23,31 @@ def test_qt_viewer(qtbot):
 
 
-def test_add_image(qtbot):
-    """Test adding image."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.random((10, 15))
-    viewer.add_image(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_volume(qtbot):
-    """Test adding volume."""
-    viewer = ViewerModel(ndisplay=3)
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.random((10, 15, 20))
-    viewer.add_image(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 3
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_pyramid(qtbot):
-    """Test adding image pyramid."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    shapes = [(40, 20), (20, 10), (10, 5)]
-    np.random.seed(0)
-    data = [np.random.random(s) for s in shapes]
-    viewer.add_image(data, is_pyramid=True)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_labels(qtbot):
-    """Test adding labels image."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = np.random.randint(20, size=(10, 15))
-    viewer.add_labels(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_points(qtbot):
-    """Test adding points."""
-    viewer = ViewerModel()
-    view = QtViewer(viewer)
-    qtbot.addWidget(view)
-
-    np.random.seed(0)
-    data = 20 * np.random.random((10, 2))
-    viewer.add_points(data)
-    assert np.all(viewer.layers[0].data == data)
-
-    assert len(viewer.layers) == 1
-    assert view.layers.vbox_layout.count() == 2 * len(viewer.layers) + 2
-
-    assert viewer.dims.ndim == 2
-    assert view.dims.nsliders == viewer.dims.ndim
-    assert np.sum(view.dims._displayed_sliders) == 0
-
-
-def test_add_vectors(qtbot):
-    """Test adding vectors."""
-    viewer = ViewerModel()
+@pytest.mark.parametrize('layer_type, data, ndim', [
```

Comment:
The data used here can actually be the same as in `test_viewer.py` and `test_napari.py`, but maybe its still better to keep this as separate file. Is there a way to define this data tuple / dict thing once in some utils file - maybe living in `napari/layers/tests/util.py` (might have to make this file) and then we can just import it once for use in multiple places?